File: packages/EvoHtmlToPdf-v4.12/Samples/NET_4.0/VB/AspNet/EvoHtmlToPdfWebDemo/DemoLinksBox.ascx.vb
Imports System.Configuration
Imports System.Collections
Imports System.Web.Security
Imports System.Web.UI
Imports System.Web.UI.WebControls
Imports System.Web.UI.WebControls.WebParts
Imports System.Web.UI.HtmlControls

Namespace EvoHtmlToPdfWebDemo
	Partial Public Class DemoLinksBox
		Inherits System.Web.UI.UserControl
		Protected Sub Page_Load(ByVal sender As Object, ByVal e As EventArgs)
		End Sub

		Public Sub LoadDemo(ByVal demoName As String)
			Select Case demoName
				Case "GettingStarted"
					lnkGettingStarted.CssClass = "demolinks selected_demolinks"
				Case "HtmlToPdfFeatures"
					lnkFeaturesDemo.CssClass = "demolinks selected_demolinks"
				Case "PdfInvoices"
					lnkPdfInvoices.CssClass = "demolinks selected_demolinks"
				Case "ConvertMultipleUrls"
					lnkConvertMultipleUrls.CssClass = "demolinks selected_demolinks"
				Case "HtmlInHeaderAndFooter"
					lnkHtmlInHeaderAndFooter.CssClass = "demolinks selected_demolinks"
				Case "HtmlElementsLocation"
					lnkHtmlElementsLocation.CssClass = "demolinks selected_demolinks"
				Case "RepeatTableHead"
					lnkRepeatTableHead.CssClass = "demolinks selected_demolinks"
				Case "TableOfContents"
					lnkTableOfContents.CssClass = "demolinks selected_demolinks"
				Case "PdfSecurity"
					lnkPdfSecurity.CssClass = "demolinks selected_demolinks"
				Case "AppendPdf"
					lnkAppendPdf.CssClass = "demolinks selected_demolinks"
				Case "HtmlToPdfElement"
					lnkHtmlToPdfElement.CssClass = "demolinks selected_demolinks"
				Case "TextAndFonts"
					lnkTextAndFonts.CssClass = "demolinks selected_demolinks"
				Case "ImageElement"
					lnkImageElement.CssClass = "demolinks selected_demolinks"
				Case "PdfShapes"
					lnkPdfShapes.CssClass = "demolinks selected_demolinks"
                Case "PdfStamps"
                    lnkPdfStamps.CssClass = "demolinks selected_demolinks"
				Case "SvgToPdf"
					lnkSvgToPdf.CssClass = "demolinks selected_demolinks"
				Case Else
			End Select
		End Sub

		Protected Sub lnkGettingStarted_Click(ByVal sender As Object, ByVal e As EventArgs)
			Response.Redirect("Default.aspx")
		End Sub

		Protected Sub lnkConvertMultipleUrls_Click(ByVal sender As Object, ByVal e As EventArgs)
			Response.Redirect("ConvertMultipleURLs.aspx")
		End Sub

		Protected Sub lnkHtmlElementsLocation_Click(ByVal sender As Object, ByVal e As EventArgs)
			Response.Redirect("HtmlElementsLocation.aspx")
		End Sub

		Protected Sub lnkRepeatTableHead_Click(ByVal sender As Object, ByVal e As EventArgs)
			Response.Redirect("RepeatTableHead.aspx")
		End Sub

		Protected Sub lnkTableOfContents_Click(ByVal sender As Object, ByVal e As EventArgs)
			Response.Redirect("TableOfContentsAndBookmarks.aspx")
		End Sub

		Protected Sub lnkPdfInvoices_Click(ByVal sender As Object, ByVal e As EventArgs)
			Response.Redirect("PdfInvoicesDemo.aspx")
		End Sub

		Protected Sub lnkPdfSecurity_Click(ByVal sender As Object, ByVal e As EventArgs)
			Response.Redirect("PdfSecurity.aspx")
		End Sub

		Protected Sub lnkAppendPdf_Click(ByVal sender As Object, ByVal e As EventArgs)
			Response.Redirect("AppendExternalPDFs.aspx")
		End Sub

		Protected Sub lnkHtmlToPdfElement_Click(ByVal sender As Object, ByVal e As EventArgs)
			Response.Redirect("HtmlToPdfElementDemo.aspx")
		End Sub

		Protected Sub lnkFeaturesDemo_Click(ByVal sender As Object, ByVal e As EventArgs)
			Response.Redirect("HtmlToPdfFeaturesDemo.aspx")
		End Sub

		Protected Sub lnkTextAndFonts_Click(ByVal sender As Object, ByVal e As EventArgs)
			Response.Redirect("TextAndFonts.aspx")
		End Sub

		Protected Sub lnkImageElement_Click(ByVal sender As Object, ByVal e As EventArgs)
			Response.Redirect("ImageElementDemo.aspx")
		End Sub

		Protected Sub lnkPdfShapes_Click(ByVal sender As Object, ByVal e As EventArgs)
			Response.Redirect("PdfShapes.aspx")
		End Sub

		Protected Sub lnkPdfStamps_Click(ByVal sender As Object, ByVal e As EventArgs)
			Response.Redirect("PdfStamps.aspx")
		End Sub

		Protected Sub lnkSvgToPdf_Click(ByVal sender As Object, ByVal e As EventArgs)
			Response.Redirect("SvgToPdfDemo.aspx")
		End Sub

		Protected Sub lnkHtmlInHeaderAndFooter_Click(ByVal sender As Object, ByVal e As EventArgs)
			Response.Redirect("HtmlInHeaderAndFooter.aspx")
		End Sub
	End Class
End Namespace

File: packages/EvoHtmlToPdf-v4.12/Samples/NET_2.0/CS/AspNet/EvoHtmlToPdfWebDemo/DemoLinksBox.ascx.cs
using System;
using System.Data;
using System.Configuration;
using System.Collections;
using System.Web;
using System.Web.Security;
using System.Web.UI;
using System.Web.UI.WebControls;
using System.Web.UI.WebControls.WebParts;
using System.Web.UI.HtmlControls;

namespace EvoHtmlToPdfWebDemo
{
    public partial class DemoLinksBox : System.Web.UI.UserControl
    {
        protected void Page_Load(object sender, EventArgs e)
        {
        }

        public void LoadDemo(string demoName)
        {
            switch (demoName)
            {
                case "GettingStarted":
                    lnkGettingStarted.CssClass = "demolinks selected_demolinks";
                    break;
                case "HtmlToPdfFeatures":
                    lnkFeaturesDemo.CssClass = "demolinks selected_demolinks";
                    break;
                case "PdfInvoices":
                    lnkPdfInvoices.CssClass = "demolinks selected_demolinks";
                    break;
                case "ConvertMultipleUrls":
                    lnkConvertMultipleUrls.CssClass = "demolinks selected_demolinks";
                    break;
                case "HtmlInHeaderAndFooter":
                    lnkHtmlInHeaderAndFooter.CssClass = "demolinks selected_demolinks";
                    break;
                case "HtmlElementsLocation":
                    lnkHtmlElementsLocation.CssClass = "demolinks selected_demolinks";
                    break;
                case "RepeatTableHead":
                    lnkRepeatTableHead.CssClass = "demolinks selected_demolinks";
                    break;
                case "TableOfContents":
                    lnkTableOfContents.CssClass = "demolinks selected_demolinks";
                    break;
                case "PdfSecurity":
                    lnkPdfSecurity.CssClass = "demolinks selected_demolinks";
                    break;
                case "AppendPdf":
                    lnkAppendPdf.CssClass = "demolinks selected_demolinks";
                    break;
                case "HtmlToPdfElement":
                    lnkHtmlToPdfElement.CssClass = "demolinks selected_demolinks";
                    break;
                case "TextAndFonts":
                    lnkTextAndFonts.CssClass = "demolinks selected_demolinks";
                    break;
                case "ImageElement":
                    lnkImageElement.CssClass = "demolinks selected_demolinks";
                    break;
                case "PdfShapes":
                    lnkPdfShapes.CssClass = "demolinks selected_demolinks";
                    break;                
                case "PdfStamps":
                    lnkPdfStamps.CssClass = "demolinks selected_demolinks";
                    break;
                case "SvgToPdf":
                    lnkSvgToPdf.CssClass = "demolinks selected_demolinks";
                    break;
                default:
                    break;
            }
        }

        protected void lnkGettingStarted_Click(object sender, EventArgs e)
        {
            Response.Redirect("Default.aspx");
        }

        protected void lnkConvertMultipleUrls_Click(object sender, EventArgs e)
        {
            Response.Redirect("ConvertMultipleURLs.aspx");
        }

        protected void lnkHtmlElementsLocation_Click(object sender, EventArgs e)
        {
            Response.Redirect("HtmlElementsLocation.aspx");
        }

        protected void lnkRepeatTableHead_Click(object sender, EventArgs e)
        {
            Response.Redirect("RepeatTableHead.aspx");
        }

        protected void lnkTableOfContents_Click(object sender, EventArgs e)
        {
            Response.Redirect("TableOfContentsAndBookmarks.aspx");
        }

        protected void lnkPdfInvoices_Click(object sender, EventArgs e)
        {
            Response.Redirect("PdfInvoicesDemo.aspx");
        }

        protected void lnkPdfSecurity_Click(object sender, EventArgs e)
        {
            Response.Redirect("PdfSecurity.aspx");
        }

        protected void lnkAppendPdf_Click(object sender, EventArgs e)
        {
            Response.Redirect("AppendExternalPDFs.aspx");
        }

        protected void lnkHtmlToPdfElement_Click(object sender, EventArgs e)
        {
            Response.Redirect("HtmlToPdfElementDemo.aspx");
        }

        protected void lnkFeaturesDemo_Click(object sender, EventArgs e)
        {
            Response.Redirect("HtmlToPdfFeaturesDemo.aspx");
        }

        protected void lnkTextAndFonts_Click(object sender, EventArgs e)
        {
            Response.Redirect("TextAndFonts.aspx");
        }

        protected void lnkImageElement_Click(object sender, EventArgs e)
        {
            Response.Redirect("ImageElementDemo.aspx");
        }

        protected void lnkPdfShapes_Click(object sender, EventArgs e)
        {
            Response.Redirect("PdfShapes.aspx");
        }
        
        protected void lnkPdfStamps_Click(object sender, EventArgs e)
        {
            Response.Redirect("PdfStamps.aspx");
        }

        protected void lnkSvgToPdf_Click(object sender, EventArgs e)
        {
            Response.Redirect("SvgToPdfDemo.aspx");
        }

        protected void lnkHtmlInHeaderAndFooter_Click(object sender, EventArgs e)
        {
            Response.Redirect("HtmlInHeaderAndFooter.aspx");
        }
    }
}